cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/CopyNativeLibTask.java

```diff
@@ -239,6 +243,42 @@ private void copyLibsFromBalo(Path baloFilePath, Set<Path> moduleDependencySet)
         }
     }
 
+    private void copyAndValidateBaloDependencies(BuildContext buildContext, Path importDependencyPath,
+                                                 PackageID packageID, Path project, Set<Path> moduleDependencySet) {
+        Manifest manifestFromBalo = RepoUtils.getManifestFromBalo(importDependencyPath);
+        List<Library> baloDependencies = manifestFromBalo.getPlatform().libraries;
+
+        if (baloDependencies == null) {
+            return;
+        }
+
+        // If platform libs are defined, copy them to target
+        List<Library> libraries = manifest.getPlatform().libraries;
+        for (Library baloTomlLib : baloDependencies) {
```

Comment:
This logic has performace impact due to loop inside loop. Can we collect all compiler libs of balos in ahash set( key can be orgname+ packageid+ artfactID+ filename) and default libs two another set. Then we can compare those two hashsets two check first one is subset of default(packing) libs.